### ISSUE TITLE
System.Numerics.Tests.Perf_VectorOf<*>.* has Regressed for Mono Interpreter Ubuntu 18.04

### ISSUE THREAD

**Issue by mrsharm** [dotnet/runtime#73863]
`System.Numerics.Tests.Perf_VectorOf<*>.*` has regressed on Mono Interpreter Ubuntu 18.04

## Mono Interpreter Ubuntu 18.04

[Reporting System](https://pvscmdupload.blob.core.windows.net/reports/allTestHistory%2frefs%2fheads%2fmain_x64_ubuntu%2018.04_LLVM%3dfalse_MonoAOT%3dfalse_MonoInterpreter%3dfalse_RunKind%3dmicro_mono%2fSystem.Numerics.Tests.Perf_VectorOf(Int32).SubtractionOperatorBenchmark.html)

![image](https://user-images.githubusercontent.com/68247673/184426921-a675a44b-724d-4c2e-be72-5885bb4581fc.png)

Repro:

```cmd
git clone https://github.com/dotnet/performance.git
py .\performance\scripts\benchmarks_monthly.py net7.0-preview7 net7.0-preview5 --filter System.Numerics.Tests.Perf_VectorOf --architecture x64
```

CC: @dakersnar

**Comment by dakersnar** [comment]
@tannergooding Considering this regression is only half a nanosecond, should we just disregard this one? That's what I'm leaning towards.

**Comment by danmoseley** [comment]
> @tannergooding Considering this regression is only half a nanosecond, should we just disregard this one? That's what I'm leaning towards.

Depending on what you're doing, it may look like 20% regression on your time?

**Comment by tannergooding** [comment]
> Normally we talk about proportions, so in this case ~20% regression. Unless it's something we expect people to not do very much, perhaps.

TL;DR: I don't think this change is particularly meaningful and we have a work item to improve these tests to track more accurate work which are less noisy and prone to mismeasurement by BDN. This is a small enough change that it's probably just a single unrelated instruction difference and not worth the time required to investigate, particularly when other SIMD workloads aren't showing meaningful regressions.

------------------------------------------

I think we need to consider the actual units of time here as well. The base unit of time for the regression, percentage of the overall regression, and likelyhood the API will be called all need to be taken into account; particularly when considering prioritizing and overall impact.

A 4GHz CPU core can theoretically execute `4` instructions per nanosecond (not including pipelining), so this `0.5ns` regression likely represents a very small difference in instruction output (possibly even just a single additional instruction). It is measuring something which is faster than even the L1 cache of a computer which means it could also be impacted by small things like code alignment differences, another service running on the computer leading to higher memory latency, etc. 

When taking that into account, and that this benchmark is effectively measuring a "single instruction", we have a consideration that the measurement is inherently somewhat noisy and that its not necessarily representative of "real world code". We also know that Benchmark.NET has some issues with measuring tests this small, hence why we have: https://github.com/dotnet/performance/issues/2555 open to ensure we can get more accurate and meaningful measurements here. Likewise, there are some known and occasionally witnessed issues with BDN when doing certain measurements. The accuracy of the high performance hardware timers is about 100ns and which querying can have a latency of up to 300ns. What this means in practice is that if we try to measure anything that takes less than 100ns of time, we will have inaccurate measurements. We want to iterate at least enough times to ensure we can get distinct start/end points and where the majority of the time between start/end is the executing code (and not things like querying the timer). I've found in practice this means we need to measure something which takes at least 1us or more, ideally it takes at least 5us to account for older hardware with higher query latencies.

I think its also worth considering that regressions less than a particular threshold are consider "same perf" already. Something that changes from `20 ms` to `21 ms` may therefore fly under the radar as "too small", but in practice is 2 million times more expensive than this benchmark is regression by. Even in the case where this operation is called 2 million times, the real world code likely has many more significant factors for improvement than this single instruction difference.

**Comment by SamMonoRT** [comment]
cc @BrzVlad
